designdocs/capacity_staking_rewards_implementation.md

```diff
@@ -0,0 +1,380 @@
+    /// The number of the last StakingEra that this account's rewards were claimed.
+    pub last_rewards_claimed_at: Option<T::RewardEra>, // NEW  None means never rewarded, Some(RewardEra) means last rewarded RewardEra.
+    /// staking amounts that have been retargeted are prevented from being retargeted again for the
+    /// configured Thawing Period number of blocks.
+    pub stake_change_unlocking: BoundedVec<UnlockChunk<BalanceOf<T>, T::RewardEra>, T::MaxUnlockingChunks>, // NEW
+}
+```
+
+### StakingTargetDetails updates, StakingHistory
+A new field, `staking_type` is added to indicate the type of staking the Account holder is doing in relation to this target.
+Staking type may be `MaximumCapacity` or `ProviderBoost`. `MaximumCapacity` is the default value for `staking_type` and maps to 0.
+
+```rust
+/// A per-reward-era record for StakingAccount total_staked amount.
+pub struct StakingHistory<Balance, RewardEra> { // NEW
+    total_staked: Balance,
+    reward_era: RewardEra,
+}
+
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+pub struct StakingTargetDetails<T: Config> {
+	/// The total amount of tokens that have been targeted to the MSA.
+	pub amount: BalanceOf<T>,
+	/// The total Capacity that an MSA received.
+	pub capacity: BalanceOf<T>,
+	/// The type of staking, which determines ultimate capacity per staked token.
+	pub staking_type: StakingType, // NEW
+    /// total staked amounts for each past era, up to StakingRewardsPastErasMax eras.
+    pub staking_history: BoundedVec<StakingHistory<BalanceOf<T>, T::RewardEra>, T::StakingRewardsPastErasMax>, // NEW
+}
+```
+
+**Unstaking thaw period**
+Changes the thaw period to begin at the first block of next RewardEra instead of immediately.
+
+### Changes to extrinsics
+#### stake
+A new parameter, `staking_type` is added to indicate the staking type. The staking type will determine how much 
+Capacity is generated for the targeted Provider. Passing `MaximumCapacity` makes the stake behave as before, whereas
+`ProviderBoost` generates a fraction of `MaximumCapacity`, splitting the staking reward as a periodically earned, 
+TBD amount of token.
+
+In the case of an increase in stake to an existing staker-target relationship, `staking_type` MUST be the same 
+as the existing relationship or else it will error with `Error::CannotChangeStakingType`.
+```rust
+pub fn stake(
+    origin: OriginFor<T>,
+    target: MessageSourceId,
+    amount: BalanceOf<T>,
+    staking_type: StakingType // NEW
+) -> DispatchResult {}
+```
+
+#### unstake
+The unstake parameters are the same, and unstake behavior is the same for `MaximumCapacity` as before, however 
+for a `ProviderBoost` staker-target relationship, the behavior must be different.  While it's not feasible to
+store neither reward_pool history nor individual staking reward history indefinitely, it still may be lengthy
```

Comment:
Double-negative "not feasilble" + "neither/nor". Either keep the "not feasible" and use "either/or", or use a positive modifier (ie, just "feasible")

Reply:
oh yeah, I reversed the meaning and forgot to update all the n's.